job-view.php

```diff
@@ -1,91 +1,122 @@
+//redirect to the home page if there is no job matching such id in the database
+if ( mysqli_num_rows( $result ) < 1 ) {
+	header( 'Location: ' . BASE_URL );
 }
 
-while ($row = mysqli_fetch_assoc($result)) :
-    ?>
-    <section class="section-fullwidth">
-        <div class="row">
-            <div class="job-single">
-                <div class="job-main">
-                    <div class="job-card">
-                        <div class="job-primary">
-                            <header class="job-header">
-                                <h2 class="job-title"><?php echo $row['title'] ?></h2>
-                                <div class="job-meta">
-                                    <a class="meta-company" href="#"><?php echo $row['company_name'] ?></a>
-                                    <span class="meta-date">
+$row = mysqli_fetch_assoc( $result );
+
+
+$job = array(
+	'title'               => '',
+	'company_name'        => '',
+	'company_description' => '',
+	'company_site'        => '',
+	'publication_date'    => '',
+	'jobs_location'       => '',
+	'salary'              => '',
+	'description'         => '',
+	'id_user'             => '',
+);
+
+foreach ( $row as $key => $value ) {
+	if ( ! empty( $value ) && array_key_exists( $key, $job ) ) {
+		$job[ $key ] = $value;
+	}
+}
+
+
+//params for include template
+$meta_title = $job['title'] . " | Jobrix.tk";
+
+//header template include
+require_once( dirname( __FILE__ ) . '/includes/theme-compat/header.php' );
+?>
+<section class="section-fullwidth">
+    <div class="row">
+        <div class="job-single">
+            <div class="job-main">
+                <div class="job-card">
+                    <div class="job-primary">
+                        <header class="job-header">
+                            <h2 class="job-title"><?php echo $job['title'] ?></h2>
+                            <div class="job-meta">
+								<?php
+								if ( ! empty( $job['company_site'] ) ) { ?>
+                                    <a class="meta-company"
+                                       href="<?php echo 'http://' . $job['company_site'] ?>"><?php echo $job['company_name'] ?></a>
+								<?php } else {
+									echo $job['company_name'];
+								}
+								?>
+                                <span class="meta-date">
                                         <?php
-                                        if (date_difference($row['publication_date']) < 31) {
-                                            echo "Posted " . date_difference($row['publication_date']) . " days ago";
+                                        if ( date_difference( $job['publication_date'] ) < 31 ) {
+	                                        echo "Posted " . date_difference( $job['publication_date'] ) . " days ago";
                                         } else {
-                                            echo "Posted over 1 month ago";
+	                                        echo "Posted over 1 month ago";
```

Comment:
why we didn't take care of this in the function and have just: `date_difference( $job['publication_date'] ` and in the function we can take care of how long the job was posted?

Reply:
fixed